File: 349-intersection-of-two-arrays/intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> st;

        for(auto it : nums1){
            st.insert(it);
        }

        vector<int> ans;
        unordered_set<int> check;
        for(auto it : nums2){
            if(st.find(it) != st.end() && !check.contains(it)){
                ans.push_back(it);
                check.insert(it);
            }
        }

        return ans;
    }
};

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {

        HashSet<Integer> st1 = new HashSet<>();
        HashSet<Integer> st2 = new HashSet<>();

        for(int num : nums1){
            st1.add(num);
        }

        for(int num : nums2){
            if(st1.contains(num)){
                st2.add(num);
            }
        }

        int[] ans = new int[st2.size()];

        int i = 0;
        for(int it : st2){
            ans[i] = it;
            i++;
        }

        return ans;
    }
}